src/core/manager.c

```diff
@@ -1995,15 +1996,33 @@ int manager_load_unit_prepare(
 
         ret = manager_get_unit(m, name);
         if (ret) {
-                *_ret = ret;
-                return 1;
+                /* The time-based cache allows to start new units without daemon-reload,
+                 * but if they are already referenced (because of dependencies or ordering)
+                 * then we have to force a load of the fragment. As an optimization, check
+                 * first if anything in the usual paths was modified since the last time
+                 * the cache was loaded. */
+                if (ret->load_state == UNIT_NOT_FOUND && m->unit_cache_mtime > 0 &&
+                                !lookup_paths_mtime_good(&m->lookup_paths, m->unit_cache_mtime)) {
+                        ret->load_state = UNIT_STUB;
+                        r = unit_load_fragment(ret);
```

Comment:
instead loading the unit here. please just reset the the type and leave it on stub as if we just allocated the unit object. i.e. leave it in a state similar to what the second part of manager_load_unit_prepare() does. i.e. this function is supposed to just *prepare* the loading, but not load itself. The loading can happen later, based on the load queue. 

i.e. follow more closely what the second part of this function does.

Reply:
Works just fine, thanks for the suggestion!